kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/ApplyTypeCheckVisitor.java

```diff
@@ -35,28 +35,30 @@ public ApplyTypeCheckVisitor(POSet<Sort> subsorts, boolean isAnywhere) {
     }
 
     public Either<java.util.Set<ParseFailedException>, Term> apply(TermCons tc) {
-        for (int i = 0, j = 0; i < tc.production().items().size(); i++) {
-            if (tc.production().items().apply(i) instanceof NonTerminal) {
-                if (tc.production().klabel().isDefined()
-                        && (tc.production().klabel().get().name().equals("#SyntacticCast")
-                        || tc.production().klabel().get().name().startsWith("#SemanticCastTo")
-                        || tc.production().klabel().get().name().equals("#InnerCast"))
-                        || (VariableTypeInferenceFilter.isFunctionRule(tc, isAnywhere) && j == 0)) {
-                    Term t = tc.get(0);
-                    boolean strict = tc.production().klabel().get().name().equals("#SyntacticCast") || tc.production().klabel().get().name().equals("#InnerCast");
-                    Either<Set<ParseFailedException>, Term> rez = new ApplyTypeCheck2(VariableTypeInferenceFilter.getSortOfCast(tc), strict).apply(t);
-                    if (rez.isLeft())
-                        return rez;
-                    tc = tc.with(0, rez.right().get());
-                    j++;
-                } else {
-                    Term t = tc.get(j);
-                    Sort s = ((NonTerminal) tc.production().items().apply(i)).sort();
-                    Either<Set<ParseFailedException>, Term> rez = new ApplyTypeCheck2(s).apply(t);
-                    if (rez.isLeft())
-                        return rez;
-                    tc = tc.with(j, rez.right().get());
-                    j++;
+        if (tc.production().params().isEmpty()) {
```

Comment:
What happens when this check fails? Should there be an assertion error? My guess is it should never happen? Or does type checking of parametric productions happen elsewhere?

Reply:
I had forgotten to rewrite this code when I made a change elsewhere, but the answer to your question is that type checking of parametric productions happens when we type check their parents.